dpe/src/commands/extend_tci.rs

```diff
@@ -29,10 +29,18 @@ impl CommandExecution for ExtendTciCmd {
         }
 
         let idx = dpe.get_active_context_pos(&self.handle, locality)?;
+        let tci_current = dpe.contexts[idx].tci.tci_current;
+        let tci_cumulative = dpe.contexts[idx].tci.tci_cumulative;
         dpe.add_tci_measurement(env, idx, &TciMeasurement(self.data), locality)?;
```

Comment:
As a general rule, it might be good to

* Make a copy of the context
* Modify it
* Set `contexts[idx] = tmp` at the end of the function

We can change `add_tci_measurement` to take a `&Context` instead of an index.

Reply:
I took a slightly different approach, where I copy out the context, and then restore contexts[idx] = copy if we run into an error after mutating contexts[idx]. The reason for this is that if we mutate the copied context instead in extend_tci for example, we would also have to change roll_onetime_use_handle to take in a context instead of an idx, and I felt this would be too invasive since it would lead to changing all calls to roll_onetime_use_handle and add_tci_measurement.

I think this way it is equally clear, but the downside is that we can't just use the ? mechanism to return.

Reply:
Disregard what I previously said, I followed your approach. I didn't want to also make roll_onetime_use_handle take in a context instead of a handle, since then we would need to update all callers of this function, so I took another approach where I mutate everything but dpe.contexts[idx].handle after we know extend_tci is sure to succeed.